nix/default.nix

```
@@ -0,0 +1,86 @@
+{
+  pkgs ? import <nixpkgs> { },
+  src ? ../.,
+  targets ? ["all"],
+  # Options: 0,1,2
+  verbosity ? 0,
+  # Use system Nim compiler instead of building it with nimbus-build-system
+  useSystemNim ? true,
+  commit ? builtins.substring 0 7 (src.rev or "dirty"),
+  # These are the only platforms tested in CI and considered stable.
+  stableSystems ? [
+    "x86_64-linux" "aarch64-linux"
+    "x86_64-darwin" "aarch64-darwin"
+  ],
+  circomCompatPkg
+}:
+
+let
+  inherit (pkgs) stdenv lib writeScriptBin callPackage;
+  
+  revision = lib.substring 0 8 (src.rev or "dirty");
+in pkgs.stdenv.mkDerivation rec {
```

Shouldn't we be enforcing GCC11?
```suggestion
in pkgs.gcc11Stdenv.mkDerivation rec {
```